pkg/build/api/types.go

```diff
@@ -231,6 +231,11 @@ type BuildOutput struct {
 	// a Docker image repository to push to. Failure to find the To will result in a build error.
 	To *kapi.ObjectReference `json:"to,omitempty"`
 
+	// pushSecretName references a name of Secret that would be used for setting
+	// up the authentication for executing the docker push to authentication
+	// enable Docker Registry (or Docker Hub).
+	PushSecretName string `json:"pushSecretName,omitempty"`
```

Comment:
@bparees thought that it would be better to be specific on what exactly we do require here ('name' of the secret)